test/sql/blob/test_blob.cpp

```diff
@@ -0,0 +1,135 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+#include "duckdb/main/prepared_statement.hpp"
+
+#include <fstream>
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Insert BLOB values from string", "[blob]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE blobs (b BLOB);"));
+	// insert BLOB from string
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('aaaaaaaaaa')"));
+	// sizes: 10, 100, 1000, 10000
+	for (idx_t i = 0; i < 3; i++) {
+		REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs SELECT b||b||b||b||b||b||b||b||b||b FROM blobs "
+								  "WHERE LENGTH(b)=(SELECT MAX(LENGTH(b)) FROM blobs)"));
+	}
+
+	result = con.Query("SELECT LENGTH(b) FROM blobs ORDER BY 1");
+	REQUIRE(CHECK_COLUMN(result, 0, {10, 100, 1000, 10000}));
+}
+
+TEST_CASE("Insert BLOB values", "[blob]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE blobs (b BLOB);"));
+	// insert BLOB
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('aaaaaaaaaa'::BLOB)"));
+	// insert BLOB with “non-printable” octets
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::BLOB)"));
+
+	// insert BLOB with “non-printable” octets, but now using VARCHAR string (should fail)
+	REQUIRE_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::VARCHAR)"));
+
+	// insert BLOB with “non-printable” octets, but now using string (should fail)
+	REQUIRE_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124')"));
+}
+
+TEST_CASE("Select BLOB values", "[blob]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE TABLE blobs (b BLOB);"));
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('aaaaaaaaaa'::BLOB)"));
+	REQUIRE_NO_FAIL(con.Query("INSERT INTO blobs VALUES ('\153\154\155 \052\251\124'::BLOB)"));
+
+	result = con.Query("SELECT count(*) FROM blobs");
```

Comment:
You should test the actual values here as well, the CHECK_COLUMN takes a list of values. Hence you can do CHECK_COLUMN(result, 0, {Value::BLOB(string(...))})

Reply:
Good catch, I will change it.